client/landing/stepper/declarative-flow/flows/onboarding-unified/onboarding-unified.ts

```diff
@@ -0,0 +1,155 @@
+		const flowName = this.name;
+		const userIsLoggedIn = useSelect(
+			( select ) => ( select( USER_STORE ) as UserSelect ).isCurrentUserLoggedIn(),
+			[]
+		);
+
+		const { setSignupDomainOrigin } = dispatch( ONBOARD_STORE ) as OnboardActions;
+		const { setShouldShowNotification } = usePurchasePlanNotification();
+
+		/**
+		 * Get post-checkout destination for onboarding-unified flow
+		 */
+		const getPostCheckoutDestination = ( providedDependencies: ProvidedDependencies ): string => {
+			// For onboarding-unified, we want to go to the site home page after setup
+			if ( providedDependencies.siteSlug ) {
+				return addQueryArgs( `/home/${ providedDependencies.siteSlug }`, { ref: flowName } );
+			}
+			// Fallback to generic home if no siteSlug
+			return '/';
+		};
+
+		/**
+		 * Handle step submissions for the onboarding unified flow
+		 */
+		const submit: SubmitHandler< typeof initialize > = async ( submittedStep ) => {
+			const { slug, providedDependencies } = submittedStep;
+
+			switch ( slug ) {
+				case 'plans': {
+					// Set domain origin as not set since this flow skips domain selection
+					setSignupDomainOrigin( SIGNUP_DOMAIN_ORIGIN.NOT_SET );
+
+					// Create minimal siteParams for post-checkout site creation
+					const siteParams = {
+						blog_name: '', // Will be auto-generated from username if empty
+						blog_title: translate( 'My Site' ), // Default site title
+						public: Visibility.PublicNotIndexed, // Coming soon by default
+						options: {
+							site_creation_flow: flowName, // Track which flow created the site
+							wpcom_public_coming_soon: 1, // Launch as coming soon
+						},
+						find_available_url: true, // Auto-find available URL since no domain was selected
+						validate: false,
+					};
+
+					// Save siteParams to localStorage for checkout to use
+					try {
+						window.localStorage.setItem( 'siteParams', JSON.stringify( siteParams ) );
+					} catch ( error ) {
+						// Silently fail if localStorage is not available
+					}
+
+					// Redirect directly to siteless checkout with selected plan
+					if ( providedDependencies?.cartItems?.length ) {
+						// Set completion tracking for post-checkout site creation
+						setSignupCompleteFlowName( flowName );
+
+						// Set the post-checkout destination to go through onboarding-unified flow steps
+						// Use our own flow's post-checkout step instead of borrowing from regular onboarding
+						persistSignupDestination( `/setup/${ flowName }/post-checkout-onboarding` );
+
+						// Get the selected plan from cartItems
+						const planItem = providedDependencies.cartItems.find( ( item ) => item.product_slug );
+
+						if ( planItem ) {
+							// Use dedicated onboarding-unified siteless checkout with plan in URL (similar to Jetpack/Akismet)
+							// Don't add signup=1 for logged-in users to avoid account creation conflicts
+							const urlParams = new URLSearchParams();
+							urlParams.set( 'flow', flowName );
```

Comment:
We should probably set the `checkoutBackUrl` param to `window.location.href.` Because currently closing the checkout by pressing X takes the user to the main onboarding flow.

![image](https://github.com/user-attachments/assets/de728a64-0fd4-4288-a6c5-6b26ea12c440)

Reply:
Agreed.
I tried setting `checkoutBackUrl` but it didn't work. I was able to get it to work using the `cancel_to` param instead.

<img width="240" alt="Screenshot 2025-07-09 at 16 56 59" src="https://github.com/user-attachments/assets/a9516699-5f4a-44be-8bba-0e810fec2089" />